finesse/hardware/opus_interface_base.py

```diff
@@ -0,0 +1,34 @@
+"""Provides a base class for interfacing with the OPUS program."""
+import logging
+import traceback
+
+from pubsub import pub
+from PySide6.QtCore import QObject
+
+
+class OPUSInterfaceBase(QObject):
```

Comment:
I guess than `QObject` and `ABC`, to make this an abstract class, do not get along and that's why you are using `NotImplementedError` below? If they do get along, I would use the abstract class and set those methods as `@abstractmethod` as that would be the canonical approach.

Reply:
Yep, that's exactly the problem. `QObject`s have their own subtle metaclass business going on. There are some workarounds but they were all pretty gnarly and didn't actually seem to work for me.